flink-runtime/src/main/java/org/apache/flink/runtime/blob/PermanentBlobService.java

```diff
@@ -50,13 +50,20 @@
     /**
      * Returns the content of the file for the BLOB with the provided job ID the blob key.
      *
+     * <p>Compared to {@code getFile}, {@code readFile} will attempt to read the entire file after
+     * retrieving it. If file reading and file retrieving is done in the same WRITE lock, it can
+     * avoid the scenario that the path to the file is deleted concurrently by other threads at the
+     * same time as the file is retrieved and read.
```

Comment:
maybe "at the same time as the file is retrieved and read" -> "when the file is retrieved but not read yet"?

Reply:
Thank you for pointing this out. Resolved.